src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -13,250 +13,198 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.databricks.spark.avro
 
-import java.io.{IOException, FileNotFoundException}
-import java.nio.ByteBuffer
-import java.util.Map
+import java.io.FileNotFoundException
+import java.util.zip.Deflater
 
+import scala.collection.Iterator
 import scala.collection.JavaConversions._
-import scala.collection.immutable.{Map => ScalaMap}
-
-import org.apache.avro.file.DataFileReader
-import org.apache.avro.generic.GenericData
-import org.apache.avro.generic.GenericData.Fixed
-import org.apache.avro.generic.{GenericRecord, GenericDatumReader}
-import org.apache.avro.mapred.FsInput
-import org.apache.avro.{SchemaBuilder, Schema}
-import org.apache.avro.Schema.Type._
-
-import org.apache.hadoop.fs.{FileSystem, Path}
-
-import org.apache.spark.rdd.RDD
-import org.apache.spark.sql.{DataFrame, Row, SQLContext}
-import org.apache.spark.sql.types._
-import org.apache.spark.sql.sources.{BaseRelation, InsertableRelation, TableScan}
-
-
-case class AvroRelation(
-    location: String,
-    userSpecifiedSchema: Option[StructType],
-    minPartitions: Int = 0,
-    recordName: String = AvroSaver.defaultParameters.get("recordName").get,
-    recordNamespace: String = AvroSaver.defaultParameters.get("recordNamespace").get)
-                       (@transient val sqlContext: SQLContext)
-  extends BaseRelation with TableScan with InsertableRelation {
-  var avroSchema: Schema = null
+import scala.collection.mutable.ArrayBuffer
+
+import com.google.common.base.Objects
+import org.apache.avro.SchemaBuilder
+import org.apache.avro.file.{DataFileConstants, DataFileReader, FileReader}
+import org.apache.avro.generic.{GenericDatumReader, GenericRecord}
+import org.apache.avro.mapred.{AvroOutputFormat, FsInput}
+import org.apache.avro.mapreduce.AvroJob
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path}
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.Logging
+import org.apache.spark.rdd.{RDD, UnionRDD}
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Row, SQLContext}
+
+abstract class AvroRelationException(msg: String) extends RuntimeException(msg)
+case class NoFilesException() extends AvroRelationException("no input files given")
```

Comment:
`AnalysisException` has a protected constructor and cannot be accessed from outside of the sql package. Is there a way around this?